packages/unchained-client/src/evm/parser/utils.ts

```diff
@@ -7,3 +10,13 @@ export const getSigHash = (inputData: string | undefined): string | undefined =>
 export const txInteractsWithContract = (tx: { to: string }, contract: string) => {
   return tx.to === contract
 }
+
+export const getTxStatus = (tx: Tx) => {
+  const status = tx.status
+
+  if (status === -1 && tx.confirmations <= 0) return TxStatus.Pending
+  if (status === 1 && tx.confirmations > 0) return TxStatus.Confirmed
+  if (status === 0) return TxStatus.Failed
+
+  return TxStatus.Unknown
+}
```

Comment:
lifted this out into a util so we can can parse tx status in swappers identically to tx history such that they will align perfectly, without the need to instantiate a tx parser or to parse a tx (which is slow for swaps as they have nested txs).